features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/CPURequirementChecker.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.listFilesSafe
+import java.io.File
+
+internal class CPURequirementChecker(
+    private val minCPUCores: Int,
+    private val cpuDirFile: File = File(DIR_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Int? = null
+
+    override fun checkMinimumRequirement(): Boolean {
+        return readCPUCoreNumber() >= minCPUCores
+    }
+
+    override fun name(): String = CPU_CHECK_NAME
+
+    override fun checkedValue(): Any? = checkedValue
```

Comment:
opinionated: I think we don't need to require the implementation to declare explicitly its generic type just because of this function, here we just need an optional value, if it returns just null we are OK with that.

Reply:
For the attributes you are sending - yes, it doesn't matter, because value type there is `Any?`, so no additional cast is needed.

But for the tests you may simplify life a bit by doing there. Also it gives a clear indicator what is the type of the metric recorded.

Anyway, not blocking to me.